File: app/src/main/java/com/artmurka/artmurkaapp/Model/InterfacesModel/ICheckoutRequest.java
package com.artmurka.artmurkaapp.Model.InterfacesModel;

import com.artmurka.artmurkaapp.Model.Pojo.ItemList.Checkout.CheckoutAllGoods;
import com.artmurka.artmurkaapp.Model.Pojo.ItemList.Categories.Success;
import com.artmurka.artmurkaapp.Model.Pojo.ItemList.CheckoutResponse.CheckoutResponse;

import io.reactivex.Observable;
import retrofit2.Call;

public interface ICheckoutRequest {
    Observable<CheckoutAllGoods> getCheckoutData();
    Call<CheckoutAllGoods> recountCheckoutData(String position, String cnt);
    Call<CheckoutResponse> postCheckout(String telephoneNumber, String adress, String email, String pay, String delivery);
}

File: app/src/main/java/com/artmurka/artmurkaapp/data/model/interfacesmodel/ICheckoutRequest.kt
package com.artmurka.artmurkaapp.data.model.interfacesmodel

import com.artmurka.artmurkaapp.data.model.pojo.itemlist.checkout.CheckoutAllGoods
import com.artmurka.artmurkaapp.data.model.pojo.itemlist.checkoutresponse.CheckoutResponse

import io.reactivex.Observable
import io.reactivex.Single
import retrofit2.Call

interface ICheckoutRequest {
    val checkoutData: Observable<CheckoutAllGoods>
    fun recountCheckoutData(position: String, cnt: String): Single<CheckoutAllGoods>
    fun postCheckout(telephoneNumber: String, adress: String, email: String, pay: String, delivery: String): Call<CheckoutResponse>
}
